src/coprocessor/dag/rpn_expr/impl_compare.rs

```diff
@@ -0,0 +1,262 @@
+            (Some(3.5), Some(-2.1), ScalarFuncSig::LTReal, Some(0)),
+            (Some(-2.1), Some(3.5), ScalarFuncSig::LTReal, Some(1)),
+            (Some(3.5), Some(3.5), ScalarFuncSig::LTReal, Some(0)),
+            (Some(-2.1), Some(-2.1), ScalarFuncSig::LTReal, Some(0)),
+            (None, None, ScalarFuncSig::LEReal, None),
+            (Some(3.5), None, ScalarFuncSig::LEReal, None),
+            (Some(-2.1), None, ScalarFuncSig::LEReal, None),
+            (None, Some(3.5), ScalarFuncSig::LEReal, None),
+            (None, Some(-2.1), ScalarFuncSig::LEReal, None),
+            (Some(3.5), Some(-2.1), ScalarFuncSig::LEReal, Some(0)),
+            (Some(-2.1), Some(3.5), ScalarFuncSig::LEReal, Some(1)),
+            (Some(3.5), Some(3.5), ScalarFuncSig::LEReal, Some(1)),
+            (Some(-2.1), Some(-2.1), ScalarFuncSig::LEReal, Some(1)),
+            (None, None, ScalarFuncSig::EQReal, None),
+            (Some(3.5), None, ScalarFuncSig::EQReal, None),
+            (Some(-2.1), None, ScalarFuncSig::EQReal, None),
+            (None, Some(3.5), ScalarFuncSig::EQReal, None),
+            (None, Some(-2.1), ScalarFuncSig::EQReal, None),
+            (Some(3.5), Some(-2.1), ScalarFuncSig::EQReal, Some(0)),
+            (Some(-2.1), Some(3.5), ScalarFuncSig::EQReal, Some(0)),
+            (Some(3.5), Some(3.5), ScalarFuncSig::EQReal, Some(1)),
+            (Some(-2.1), Some(-2.1), ScalarFuncSig::EQReal, Some(1)),
+            (None, None, ScalarFuncSig::NEReal, None),
+            (Some(3.5), None, ScalarFuncSig::NEReal, None),
+            (Some(-2.1), None, ScalarFuncSig::NEReal, None),
+            (None, Some(3.5), ScalarFuncSig::NEReal, None),
+            (None, Some(-2.1), ScalarFuncSig::NEReal, None),
+            (Some(3.5), Some(-2.1), ScalarFuncSig::NEReal, Some(1)),
+            (Some(-2.1), Some(3.5), ScalarFuncSig::NEReal, Some(1)),
+            (Some(3.5), Some(3.5), ScalarFuncSig::NEReal, Some(0)),
+            (Some(-2.1), Some(-2.1), ScalarFuncSig::NEReal, Some(0)),
+            (None, None, ScalarFuncSig::NullEQReal, Some(1)),
+            (Some(3.5), None, ScalarFuncSig::NullEQReal, Some(0)),
+            (Some(-2.1), None, ScalarFuncSig::NullEQReal, Some(0)),
+            (None, Some(3.5), ScalarFuncSig::NullEQReal, Some(0)),
+            (None, Some(-2.1), ScalarFuncSig::NullEQReal, Some(0)),
+            (Some(3.5), Some(-2.1), ScalarFuncSig::NullEQReal, Some(0)),
+            (Some(-2.1), Some(3.5), ScalarFuncSig::NullEQReal, Some(0)),
+            (Some(3.5), Some(3.5), ScalarFuncSig::NullEQReal, Some(1)),
+            (Some(-2.1), Some(-2.1), ScalarFuncSig::NullEQReal, Some(1)),
+        ];
+        for (arg0, arg1, sig, expect_output) in test_cases {
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(arg0)
+                .push_param(arg1)
+                .evaluate(sig)
+                .unwrap();
+            assert_eq!(output, expect_output);
```

Comment:
Give a context when `assert_eq!` failed, e.g.

```suggestion
            assert_eq!(output, expect_output, "{:?}, {:?}, {:?}", arg0, arg1, sig);
```